cohortextractor2/dsl.py

```diff
@@ -161,9 +169,109 @@ class PatientSeries:
     def __init__(self, value: Value):
         self.value = value
 
+    @staticmethod
+    def _get_other(other: Expression) -> Comparator | Value:
+        if isinstance(other, PatientSeries):
+            return other.value
+        return other
 
-Expression = Union[PatientSeries, bool, int, str]
+    def __gt__(self, other: Expression) -> Comparator:
+        return self.value > self._get_other(other)
 
+    def __ge__(self, other: Expression) -> Comparator:
+        return self.value >= self._get_other(other)
 
-def categorise(mapping: dict[str, Expression]) -> PatientSeries:
-    """Represents a switch statement."""
+    def __lt__(self, other: Expression) -> Comparator:
+        return self.value < self._get_other(other)
+
+    def __le__(self, other: Expression) -> Comparator:
+        return self.value <= self._get_other(other)
+
+    def __eq__(self, other: Expression) -> Comparator:  # type: ignore[override]
```

Comment:
What's the reason for the ignore here? I feel we ought to be able to type this correctly.

Reply:
mypy complains because `__eq__` should return a bool in the thing it's overriding (object), and it says you shouldn't return a narrower return type than the thing being overridden

Reply:
Ooooh, I suppose it's because _all_ objects have `__eq__` defined. So we can get away with typing the other dunder methods "incorrectly" because we're not overriding anything. Python's really not build for being accurately typed, is it? Thanks. Any chance of a comment to explain?

Reply:
Done :)